uint16 number_of_points
---
bool success
string message